File: UserDev/RecoTool/CMTool/CMToolApp/arxiv/bin/test_merge.cc
#include "Analysis-TypeDef.hh"
#include "CMTool-TypeDef.hh"
#include "TSystem.h"

int main(int argc, char** argv) {

  larlight::ana_processor mgr;

  mgr.add_input_file("../../../data/electron_00/combined.root");
    
  mgr.set_input_rootdir("scanner");

  mgr.set_output_file("");

  mgr.set_io_mode(larlight::storage_manager::READ);
  
  mgr.set_ana_output_file("");
  
  larlight::ClusterMerger* proc = new larlight::ClusterMerger;;

  //proc->GetManager().AddMergeAlgo(new cluster::CMAlgoFake());

  //proc->GetManager().AddSeparateAlgo(new cluster::CMAlgoFake());

  proc->GetManager().AddMergeAlgo(new cmtool::CBAlgoMergeAll);

  proc->GetManager().DebugMode(cmtool::CMergeManager::kPerIteration);
  //proc.GetManager().DebugMode(cluster.CMergeManager.kPerAlgoSet);
  //proc.GetManager().DebugMode(cluster.CMergeManager.kPerMerging);

  mgr.add_process(proc);

  mgr.run(0,5000);

  return 1;
}

File: UserDev/RecoTool/CMTool/CMTAlgMerge/corey_merging/MergingScripts/test_merge.py
import ROOT,sys
ROOT.gSystem.Load("libClusterRecoUtil")
from ROOT import *
from ROOT import larlight as fmwk

mgr = fmwk.ana_processor()

mgr.add_input_file(sys.argv[1])

mgr.set_input_rootdir("scanner")

mgr.set_output_file("")

mgr.set_io_mode(fmwk.storage_manager.READ)

mgr.set_ana_output_file("")

proc = fmwk.ClusterMerger()

proc.GetManager().AddMergeAlgo(cmtool.CBAlgoFake())

proc.GetManager().AddSeparateAlgo(cmtool.CBAlgoFake())

proc.GetManager().DebugMode(cmtool.CMergeManager.kPerIteration)
#proc.GetManager().DebugMode(cmtool.CMergeManager.kPerAlgoSet)
#proc.GetManager().DebugMode(cmtool.CMergeManager.kPerMerging)

mgr.add_process(proc)

while mgr.process_event():

    print() 
    bk=proc.GetManager().GetBookKeeper()

    res = bk.GetResult()

    for x in range(res.size()):

        for y in range(res.at(x).size()):

            print(res.at(x).at(y), end=' ')
        print()

    sys.stdin.readline()



